executor.go

```diff
@@ -397,7 +397,10 @@ func doesFragmentConditionMatch(eCtx *ExecutionContext, fragment ast.Node, ttype
 		if conditionalType == ttype {
 			return true
 		}
-
+                if conditionalType.Name() == ttype.Name() {
```

Comment:
I guess its not absolutely necessary but since every type has to have a unique name this should also hold valid true. It's very useful in because say we have a type ex:
`type Quiz struct {}` which I use internally I want to be able to send some of its data back to client without creating a new struct so we use a map[string]interface{} and map out the fields we want to send back but the problem arises when using relay.

When we are using nodeDefinition's it doesn't know it is a quiz type but is map[string]interface{}so for this we use the type name like this
Ofcourse we can use any other field but this is easier to reason and it adds some sort of dynamic behavoir to the strong go type struct type system.
```go
MutateAndGetPayload(params graphql.Params) (interface{}, error) {
  return map[string]interface{}{
    "__typename": "Quiz",
    "id": "quiz1",
  }
}

//and in our nodeDefinitions we resolve the type like this
NodeDefinitions = relay.NewNodeDefinitions(relay.NodeDefinitionsConfig{
	IDFetcher: func(id string, info graphql.ResolveInfo) interface{} {
		resolve := relay.FromGlobalID(id)
		switch resolve.Type {
		case "Quiz":
			return map[string]interface{}{
				"__typename": "Quiz",
				"id": "some_id",
			}
            }
    }
  },
            }
  TypeResolve: func(value interface{}, info graphql.ResolveInfo) *graphql.Object {
		switch data := value.(type) {
		case map[string]interface{}:
			switch data["__typename"] {
			case "Quiz":
				return QuizType
			}
		}
	}
})
```
This helps us in not creating to many structs for a single graphql Type.